docs/adr/2023-08-18_025-resource-based-api.md

```diff
@@ -0,0 +1,145 @@
+      resource-specific _commands_. Any commands accepted must also be available
+      via the corresponding websocket connection.
+
+  - `Accept` request headers can be used to configure the `Content-Type` of the
+    response
+
+    - All resources must provide `application/json` responses
+
+    - Some resources might support more content types (e.g. CBOR-encoded binary)
+
+  - Query parameters may be used to further configure responses of some
+    resources. For example, `?address=<bech32>` could be used to filter UTxO by
+    some address.
+
+- Keep the semantics of `/`, which accepts websocket upgrade connections and
+  sends direct/raw output of `ServerOutput` events on `/`, while accepting all
+  `ClientCommand` messages.
+
+  - Define `ServerOutput` also in terms of the `StateChanged` event stream
+
+### Example resources
+
+Example resource paths + HTTP verbs mapped to existing things to demonstrate the
+effects of the decision points above. The mappings may change and are to be
+documented by an API specification instead.
+
+| Path                             | GET                            | POST                   | PATCH   | DELETE             |
+| :------------------------------- | :----------------------------- | :--------------------- | ------- | :----------------- |
+| `/v1/head/status`                | `HeadStatus(..)`               | -                      | -       | -                  |
+| `/v1/head/snapshot/utxo`         | last confirmed snapshot utxo   | -                      | -       | -                  |
+| `/v1/head/snapshot/transactions` | confirmed snapshot txs         | `NewTx` + responses    | -       | -                  |
+| `/v1/head/ledger/utxo`           | `localUTxO`                    | -                      | -       | -                  |
+| `/v1/head/ledger/transactions`   | `localTxs`                     | `NewTx` + responses    | -       | -                  |
+| `/v1/head/commit`                | -                              | `Chain{draftCommitTx}` | -       | -                  |
+| `/v1/head`                       | all `/v1/head/*` data          | `Init`                 | `Close` | `Fanout` / `Abort` |
+| `/v1/protocol-parameters`        | current protocol parameters    |                        |         |                    |
+| `/v1/cardano-transaction`        | -                              | `Chain{submitTx}`      | -       | -                  |
+| `/v1/peers`                      | a list of peers                | -                      | -       | -                  |
+| `/v1/node-version`               | node version as in `Greetings` | -                      | -       | -                  |
+| `/v1/`                           | all `/v1/*` data               | -                      | -       | -                  |
+
+Multiple heads are out of scope now and hence paths are not including a
+`<headId>` variable section.
```

Comment:
As this is an ADR, and not an implementation plan, I would like to suggest we include the `headId` in the path to queries retrieving the state of a particular head. I think multiple heads are not that far down the road, this does not add much complexity, and seems to be more consistent with traditional practices. 
I am fine with leaving it as it is if you think it's not worth the hassle.

Reply:
I'd like to keep it out for now.